FaunaDB.Client/Client/HttpRequestExtensions.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Net.Http;
+namespace FaunaDB.Client
+{
+    public static class HttpRequestExtensions
+    {
+        private static string TimeoutPropertyKey = "RequestTimeout";
```

Comment:
Where did you find oficial documentation about this property? Microsoft documentation don't mention anything about it https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httprequestmessage.properties?view=netframework-4.8

Reply:
https://thomaslevesque.com/2018/02/25/better-timeout-handling-with-httpclient/

Reply:
this is another alternative to set cancellation token https://stackoverflow.com/questions/28060575/how-to-set-time-out-for-http-client-request-operation-in-windows-phone-8-1-windo

Reply:
I can add the cancellation token too if that's better

Reply:
looks like `request.Properties` is not used in any place on the source code https://github.com/microsoft/referencesource/tree/master/System/net/System/Net/Http

Reply:
looks like the best thing to do is this. It will cancel the request when the token expires causing it timeout 
```
CancellationTokenSource cts = new CancellationTokenSource(2000); // 2 seconds
    HttpClient client = new HttpClient();
    HttpResponseMessage response = await
        client.SendRequestAsync(request).AsTask(cts.Token);
```

Reply:
I think we could remove the extension, and add the timeout in the client creation... I can confirm that this patch works:
```
diff --git a/FaunaDB.Client/Client/DefaultClientIO.cs b/FaunaDB.Client/Client/DefaultClientIO.cs
index 7e1f3b7..0e5fd0c 100644
--- a/FaunaDB.Client/Client/DefaultClientIO.cs
+++ b/FaunaDB.Client/Client/DefaultClientIO.cs
@@ -35,7 +35,7 @@ namespace FaunaDB.Client
         }

         public DefaultClientIO(string secret, Uri endpoint, TimeSpan timeout, HttpClient httpClient = null)
-            : this(httpClient ?? CreateClient(), AuthHeader(secret), new LastSeen(), endpoint, timeout)
+            : this(httpClient ?? CreateClient(timeout), AuthHeader(secret), new LastSeen(), endpoint, timeout)
         { }

         public IClientIO NewSessionClient(string secret) =>
@@ -127,9 +127,14 @@ namespace FaunaDB.Client
             return string.Join("&", keyValues);
         }

-        static HttpClient CreateClient()
+        static HttpClient CreateClient(TimeSpan? timeout)
         {
-            return new HttpClient();
+            var c = new HttpClient();
+
+            if (timeout.HasValue)
+                c.Timeout = timeout.Value;
+
+            return c;
         }
     }
 }```

Reply:
Ok... you can remove every timeout related code, we are going to manage this in the server side.

Reply:
I assume you'll need to send timeout in some header or something so ill just remove the extensions and SetTimeout() call in DefaultClientIO and leave timeout in the DefaultClientIO constructor?

Reply:
yeah, that's the idea, the driver will send a header with the desired timeout per query and it will control the entire pipeline on the server side

Reply:
ok so the new commit should fix that you guys just have to add the header in a separate commit

Reply:
if you want to give me the header key I can set it to the timeout variable

Reply:
Before we move to this new header approach, I would like to finish and merge this PR. Could  you do it ? We just need to remove the current timeout related stuff and this PR is good to go!

Reply:
In the meantime I'm looking for the header specs (such as, is this milliseconds or seconds?)

Reply:
Wouldn't the best way to do it server side is have two headers one is an int. The other is an enum with values mil, sec, tick etc. Then it doesn't matter what the clients send users can send whatever they want and server side figures out the conversions

Reply:
I don't off the top of my head know what C# timeout is bc I always leave it default

Reply:
I removed the timeout stuff too in commit f0a1c92. It still exists in the class ready to be put in a header

Reply:
the client uses TimeSpan so users can set it to anything TimeSpan.FromSeconds()/Hours() etc

Reply:
you can do this on the client timeout.TotalMilliseconds to then pass it to the header

Reply:
I'm not sure YET how the header is going to behave but using TimeSpan to convert to milliseconds or seconds sounds really nice (user friendly)